### PR TITLE
bug: Fix auth header for push

### LINKED ISSUE #1079: Fix AUTH header for push component
The auth header is missing the `webpush` type info, causing it to fail on the server.

### PR DESCRIPTION
**Pull request by jrconlin**
* include auth type prefix
* fix URL for channel list (note:
  https://github.com/mozilla-services/autopush/issues/1330

Closes #1079

### Pull Request checklist ###

- [x] **Quality**: This PR builds and tests run cleanly
  - `cargo test --all` produces no test failures
  - `cargo clippy --all --all-targets --all-features` runs without emitting any warnings
  - `cargo fmt` does not produce any changes to the code
  - `./gradlew ktlint detekt` runs without emitting any warnings
  - Note: For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes a changelog entry or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [x] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/master/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

### CODE REVIEW

File: components/push/src/communications.rs [hunk 1]
```diff
@@ -196,9 +199,12 @@ impl Connection for ConnectHttp {
             if requested.status == status_codes::CONFLICT {
                 return Err(AlreadyRegisteredError.into());
             }
-            return Err(
-                CommunicationError(format!("Unhandled client error {:?}", requested)).into(),
-            );
+            return Err(CommunicationError(format!(
+                "Unhandled client error {} : {:?}",
+                requested.status,
+                String::from_utf8_lossy(&requested.body)
```
Comment (thomcc) [inline components/push/src/communications.rs]: Hmm, is there any risk of PII in the body?

File: components/push/src/communications.rs [hunk 2]
```diff
@@ -209,8 +215,13 @@ impl Connection for ConnectHttp {
...
         };
 
-        self.uaid = response["uaid"].as_str().map(ToString::to_string);
-        self.auth = response["secret"].as_str().map(ToString::to_string);
+        if self.uaid.is_none() {
+            self.uaid = response["uaid"].as_str().map(ToString::to_string);
+        }
+        if self.auth.is_none() {
+            self.auth = response["secret"].as_str().map(ToString::to_string);
+            println!("### Got Auth {:?}", self.auth);
```
Comment (thomcc) [inline components/push/src/communications.rs]: Nit: get rid of the println before landing.

Comment (jrconlin) [reply components/push/src/communications.rs]: ah, crap! Thanks!

File: components/push/src/communications.rs (hunk 1 above)
Comment (jrconlin) [reply components/push/src/communications.rs]: Quite the opposite. Our push servers return [extended help info](https://autopush.readthedocs.io/en/latest/http.html?highlight=error#error-resp) whenever they hit an error.  This info was getting lost in how the message was being returned.

**Pull request merged**